src/core/Note.cpp

```diff
@@ -237,4 +237,33 @@ bool Note::withinRange(int tickStart, int tickEnd) const
 }
 
 
+
+
+/*! \brief Get the start/end/bottom/top positions of notes in a vector
+ *
+ *  Returns no value if there are no notes
+ */
+std::optional<NoteBounds> boundsForNotes(const NoteVector& notes)
+{
+	if (notes.empty()) { return {}; }
```

Comment:
I wonder whether `return std::nullopt;` would be clearer here.

Reply:
I thought `{}` was enough, but `std::nullopt` looks clearer to me.

Reply:
I decided to drop `std::optional` and go with `operator bool()` to return true if the NoteBound is non-zero. That way we get return value optimization like @JohannesLorenz in a clean and simple way. Thoughts?